### DISCUSSION TITLE
Welcome to pyglider Discussions!

### DISCUSSION THREAD

**Discussion by jklymak** [c-proof/pyglider#16] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by soerenthomsen** [comment]
Welcome, I am Soeren Thomsen, Best Practice coordinator at OceanGliders and also engaged with GliderTools. Glad to see activity and @callumrollo here. 

Are there plans to make these tools interoperable with GliderTools and adapting the new OG1.0 format?

**Reply by callumrollo** [reply]
Hi Soeren! I was wondering when you'd pop up :P 

We were just discussing the direction we'd like this package to take. We plan to work toward the OG 1.0 format, but it'll be a ways down the road. I think OG 1.0 would be an ideal exchange mechanism between pyglider and GliderTools. pyglider will most likely remain low level, extracting data from slocum/seaexplorer format and making metadata-rich netcdfs. These should ideally be usable as input to GliderTools to make the most of the higher level processsing (oxygen correction, quenching etc.) and more sophisticated plotting.

I believe GliderTools already has plans to support OG 1.0 as an input format once it is finalised?

**Reply by jklymak** [reply]
Yes, the end goal is og1.0 format. We just haven't put the work into doing that until the standard is done.  At some point we _were_ compatible with GDAC at OOI, but we may have drifted.  

Last time I looked at glider tools it was basically format agnostic, and did all its work on numpy vectors.  If so, then we are already compatible!

**Reply by soerenthomsen** [reply]
Thanks for the quick replies and sounds all great! Low level data extracting for slocum is a key missing part for GliderTools as I understand at the moment. So coming together here would be amazing. 

Yes GliderTools plans to support OG1.0 here the issues: https://github.com/GliderToolsCommunity/GliderTools/issues/49 and https://github.com/GliderToolsCommunity/GliderTools/issues/50 

At the moment I am very occupied with moving all OceanGliders SOPs to GitHub for community review. The SOPs will provide the state of the art knowledge for improving high level processing of GliderTools i.e. oxygen will be done next:

- The oxygen SOP is already online [here](https://oceangliderscommunity.github.io/Oxygen_SOP/README.html) and open for community review
- Salinity, Nitrate and Depth Average Currents will be uploaded in the next weeks/months to the OceanGlidersCommunity GitHub repo

I am happy to coordinate a synchronous discussion call with you guys and active people of GliderTools whenever it makes sense. Maybe later this year?

**Reply by callumrollo** [reply]
Sounds good man. It will take a good month to get pyglider overhauled and work toward OG 1.0 format. A meeting later in the year to clarify what processed should be in which package and avoid duplication of effort would be good

**Reply by soerenthomsen** [reply]
Just ping me @callumrollo when it makes sense and I will take the lead to organise it.

**Comment by callumrollo** [comment]
Hi all. I'm Callum Rollo, data scientist/physical oceanographer at [VOTO](https://voiceoftheocean.org/). I'm new to SeaExplorer data, but have 4 years of experience with Seagliders. 

I aim to improve the flexibility of pyglider's SeaExplorer processing to handle a wide range of sensors. I hope to use my experience packaging other Python projects to eventually get pyglider onto PyPI/conda-forge

**Comment by richardsc** [comment]
Hi! I'm Clark Richards, a Research Scientist at Fisheries and Oceans Canada and the co-lead of the "Coastal Ocean Glider Group" at the Bedford Institute of Oceanography in Halifax Canada. 

We have a fleet of 5 SeaExplorer gliders, and are likely to add some slocum gliders in the near future. We have an array of payloads, including GPCTDs, O2 (SBE and Rinko), an RBRlegato CTD, FLBBCD EcoPuck, and a hydrophone. I'm keen to see `pyglider` get utilized within Canada for a nicely coordinated glider data management system, as well as beyond!

**Comment by Kikelon** [comment]
Hi! I'm Enrique Castrillo, a Computer Engineer that has just arrived at SOCIB and is getting inside this marine world. I'm starting with Slocum Gliders and Python at my current position, and just found this project that seems quite interesting. I'll love to learn from all of you as much as I can.

Nice to meet you!

**Reply by jklymak** [reply]
Welcome @Kikelon 

Let us know if you have any questions.  Also note that we don't have access to every glider type, so if there needs to be special integration w your instruments, please let us know, and/or share any modifications you find are needed!

**Reply by ecastrillo** [reply]
Sorry, I'm Kikelon, but I didn't realize that I posted here with my personal account... 

Thank you for the wellcome. I'll try to contribute as much as I can been a newbie with Python, NetCDFs, Gliders, etc... 

My current goal is to read from raw files and generate a OG1.0 format from [Ocean Gliders Community](https://github.com/OceanGlidersCommunity) using Python. For this reason, I wanted to use pyglider to achieve the first step (Raw -> NetCDF).

**Reply by jklymak** [reply]
Yes, we don't make OG1.0 yet because IOOS GDAC is still using an older format.  I'm curious if they will start accepting the OG1.0 format, or if we will have two formats for a while.  Writing a converter to OG1.0 should be easy enough.
